x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/action/TransportMountSearchableSnapshotAction.java

```diff
@@ -132,6 +133,11 @@ protected void masterOperation(
     ) {
         SearchableSnapshots.ensureValidLicense(licenseState);
 
+        final String mountedIndexName = request.mountedIndexName();
+        if (mountedIndexName.charAt(0) == '.') {
```

Comment:
I think it would be best to pull in the `SystemIndices` class since we're in a transport action. We can bind the instance in `Node` and make it available for guice to inject. Then a simple modification to the the `SystemIndices` class could be made that adds a method to check the name (currently there is a method that takes an `Index` object but only looks at the name) against the defined set of system indices.

The primary reason for this is users could have data indices in 7.x that start with a `.` so we shouldn't break that for them.

Reply:
There are also hidden indices that start with a dot (e.g. `.watcher-history*`), which there's no reason to prevent being backed by a searchable snapshot. Using the `SystemIndices` class would avoid that issue as well.

Reply:
I merged #61540 which adds the method in `SystemIndices` to check a string name without needing the `Index` object.

Reply:
Thanks everybody. I'm waiting for #60522 to be merged and I'll update this PR.